### ISSUE TITLE
FR: toolbar button to disable chutzpah's test discovery (was: VS2017 is telling me Chutzpah is a startup performance problem)

### ISSUE THREAD

**Issue by gregveres** [mmanela/chutzpah#671]
When I launch VS 2017 (something happened in 2015), I can't run my unit tests for quite a while - maybe a minute or so. The green cylon bar just keeps going across the top of the test explorer. Then finally it stops and I can run it. 

I believe what is happening is that the Chutzpah test runner is scanning for the tests in all my JS files. I have about 5000 unit tests. 

I don't think this initial scan does anything useful. Sure it populates the list of unit tests in the test explorer but with 10,000 unit tests in there (another 5,000 for C#) the list of not run tests is of no value. And I think it does the scan all over again when I do ask it to run all the unit tests. 

Is this what is happening? Is Chutzpah running the scan on startup? Is there a way we can prevent that?

**Comment by mmanela** [comment]
Chutzpah is scanning at startup. This is how the testexplorer framework wants it to act. If you don't tell the test explorer framework that you have tests it won't let you run them! I agree its an issue and I am looking for ways to improve it or make it faster.

**Comment by gregveres** [comment]
Hi @mmanela 

I started to disable Chutzpah's test adaptor extension when I am doing my C# unit tests. I work on the back end and the front end but when I do back end work, I generally focus on building out the APIs with unit tests in C#. I have noticed that VS 2017 performs SO much faster doing C# unit tests when the chutzpah test adaptor extension is disabled so I have started to disable it before doing my C# work and then re-enabling it when I am done. 

Would it be possible to have a setting somewhere that tells Chutzpah to just return no tests found and skips the test discovery? Then when the setting is off it does the full scan and returns the full list of tests?

Having something like that would save me a ton of wasted time waiting for VS to finally be ready to run my unit tests. I would love to be able to put a toggle button on the toolbar so that I could just hit the button to disable chutzpah (so I don't have restart VS), then when I need to run my typescript unit tests, I hit the toggle button again and I am running all my unit tests.
